### PR TITLE
feature: module-5 template implemented

### PR DESCRIPTION
**Pull request by auchynnikau**

### CODE REVIEW

File: angular.json [hunk 1]
```diff
@@ -120,6 +122,7 @@
           }
         }
       }
-    }},
+    }
+  },
   "defaultProject": "angular-gmp"
-}
+}
```
Comment (katrin2404) [inline angular.json:128]: Run linter, please.

File: src/app/modules/courses/components/courses-list/courses-list.component.ts [hunk 2]
```diff
@@ -1,34 +1,45 @@
...
 import { CourseProps } from '../../../../shared/models/course';
 import { FilterPipe } from '../../../../shared/pipes/filter.pipe';
+import { CoursesService } from '../../services/courses.service';
 
 @Component({
   selector: 'vc-courses-list',
   templateUrl: './courses-list.component.html',
   styleUrls: ['./courses-list.component.scss'],
-  providers: [FilterPipe],
+  providers: [FilterPipe, CoursesService],
```
Comment (katrin2404) [inline src/app/modules/courses/components/courses-list/courses-list.component.ts:10]: Why did you declare CoursesService in component?

File: src/app/modules/courses/services/courses.service.ts [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  getCoursesItem(id: string): CourseProps {
+    return this.courses.find(({ id: courseId }) => courseId === id);
+  }
+
+  createCourse(course: CourseProps): void {
+    this.courses.push(course);
+  }
+
+  updateCourse(course: CourseProps): void {
+    const courseToUpdate = this.courses.findIndex(({ id }) => id === course.id);
```
Comment (katrin2404) [inline src/app/modules/courses/services/courses.service.ts:20]: Is `updatedCourseIndex ` better?

File: src/app/modules/login/login.module.ts [hunk 4]
```diff
@@ -0,0 +1,14 @@
+import { NgModule } from '@angular/core';
+import { FormsModule } from '@angular/forms';
+import { BrowserModule } from '@angular/platform-browser';
+import { LoginComponent } from './pages/login/login.component';
+import { SharedModule } from '../../shared/shared.module';
+import { LoginFormComponent } from './components/login-form/login-form.component';
+
+@NgModule({
+  imports: [BrowserModule, SharedModule, FormsModule],
+  declarations: [LoginFormComponent],
```
Comment (katrin2404) [inline src/app/modules/login/login.module.ts:10]: Declare LoginComponent. All components should be declare in some module.

File: src/app/modules/login/pages/login/login.component.html [hunk 5]
```diff
@@ -0,0 +1,3 @@
+<!-- <vc-header></vc-header> -->
```
Comment (katrin2404) [inline src/app/modules/login/pages/login/login.component.html:1]: Why did you comment header and footer?

File: src/app/shared/components/header/header.component.html [hunk 6]
```diff
@@ -1,4 +1,4 @@
 <header class="header">
   <vc-logo></vc-logo>
-  <vc-user-info></vc-user-info>
+  <vc-user-info *ngIf="isAuthenticated" (logout)="logout()"></vc-user-info>
```
Comment (katrin2404) [inline src/app/shared/components/header/header.component.html:3]: I recommend organizing your layout in such a way that those components that do not depend on authentication are always visible, the list of courses and the login page are displayed depending on whether the user is authenticated or not.

File: src/app/shared/components/header/header.component.ts [hunk 7]
```diff
@@ -1,8 +1,19 @@
...
   selector: 'vc-header',
   templateUrl: './header.component.html',
   styleUrls: ['./header.component.scss'],
+  providers: [AuthService],
 })
-export class HeaderComponent {}
+export class HeaderComponent {
+  constructor(private authService: AuthService) {}
+
+  isAuthenticated: boolean = this.authService.checkIsAuthenticated();
```
Comment (katrin2404) [inline src/app/shared/components/header/header.component.ts:13]: In this case you get data one time.

File: src/app/shared/mocks/user.ts [hunk 8]
```diff
@@ -0,0 +1,7 @@
+import { UserProps } from '../models/user';
+
+export const userMocks: UserProps = {
+  id: '123',
+  firstName: 'Steve',
+  lastName: 'Martin',
```
Comment (katrin2404) [inline src/app/shared/mocks/user.ts:6]: I offer to add userName and password here and when user login in app to match userName + password and return token.

File: src/app/shared/services/auth.service.ts [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+  private isAuthenticated: boolean = this.checkIsAuthenticated();
+
+  private user: UserProps = { ...userMocks };
+
+  private token = '123e143245t';
+
+  login(): void {
+    const { firstName, lastName } = this.user;
+    localStorage.setItem('user', `${firstName} ${lastName}`);
+    localStorage.setItem('token', this.token);
```
Comment (katrin2404) [inline src/app/shared/services/auth.service.ts:14]: It's better to provide userName and password here and check in userMock

File: src/app/modules/courses/pages/courses/courses.component.ts [hunk 10]
```diff
@@ -1,12 +1,17 @@
...
   selector: 'vc-courses',
   templateUrl: './courses.component.html',
   styleUrls: [],
+  providers: [AuthService],
 })
 export class CoursesComponent {
+  constructor(private authService: AuthService) {}
+
   searchQuery: string;
+  isAuthenticated: boolean = this.authService.checkIsAuthenticated();
```
Comment (katrin2404) [inline src/app/modules/courses/pages/courses/courses.component.ts:14]: In this case you get data just on init component.

File: src/app/modules/login/components/login-form/login-form.component.html [hunk 11]
```diff
@@ -0,0 +1,22 @@
...
+    <mat-form-field appearance="fill">
+      <mat-label>Enter your e-mail</mat-label>
+      <input matInput [(ngModel)]="email">
+    </mat-form-field>
+  </div>
+  <div>
+    <mat-form-field appearance="fill">
+      <mat-label>Enter your password</mat-label>
+      <input matInput [(ngModel)]="password" [type]="hide ? 'password' : 'text'">
+      <button mat-icon-button matSuffix (click)="hide = !hide" [attr.aria-label]="'Hide password'" [attr.aria-pressed]="hide">
```
Comment (katrin2404) [inline src/app/modules/login/components/login-form/login-form.component.html:13]: I don't see `hide` property in your component.

File: angular.json (hunk 1 above)
Comment (auchynnikau) [reply angular.json:128]: Fixed.

File: src/app/modules/courses/components/courses-list/courses-list.component.ts (hunk 2 above)
Comment (auchynnikau) [reply src/app/modules/courses/components/courses-list/courses-list.component.ts:10]: To have an ability to use service functionality, without that record course list component wouldn't work correctly.

File: src/app/modules/courses/services/courses.service.ts (hunk 3 above)
Comment (auchynnikau) [reply src/app/modules/courses/services/courses.service.ts:20]: Sure, thanks.

File: src/app/modules/login/login.module.ts (hunk 4 above)
Comment (auchynnikau) [reply src/app/modules/login/login.module.ts:10]: Fixed.

File: src/app/modules/login/pages/login/login.component.html (hunk 5 above)
Comment (auchynnikau) [reply src/app/modules/login/pages/login/login.component.html:1]: Fixed.

File: src/app/shared/components/header/header.component.html (hunk 6 above)
Comment (auchynnikau) [reply src/app/shared/components/header/header.component.html:3]: Done.

File: src/app/modules/login/components/login-form/login-form.component.html (hunk 11 above)
Comment (auchynnikau) [reply src/app/modules/login/components/login-form/login-form.component.html:13]: Fixed.

File: src/app/shared/components/header/header.component.ts (hunk 7 above)
Comment (auchynnikau) [reply src/app/shared/components/header/header.component.ts:13]: Fixed.

File: src/app/modules/courses/pages/courses/courses.component.ts (hunk 10 above)
Comment (auchynnikau) [reply src/app/modules/courses/pages/courses/courses.component.ts:14]: Fixed.

File: src/app/shared/mocks/user.ts (hunk 8 above)
Comment (auchynnikau) [reply src/app/shared/mocks/user.ts:6]: Done.

File: src/app/shared/services/auth.service.ts (hunk 9 above)
Comment (auchynnikau) [reply src/app/shared/services/auth.service.ts:14]: Done.

File: src/app/modules/courses/components/courses-list/courses-list.component.ts (hunk 2 above)
Comment (katrin2404) [reply src/app/modules/courses/components/courses-list/courses-list.component.ts:12]: It's better to register service in module.

**Review by katrin2404 — APPROVED** [review]

**Pull request merged**